akka-http-core/src/test/scala/akka/http/scaladsl/SslConfigWarningsSpec.scala

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.scaladsl
+
+import akka.actor.ActorSystem
+import akka.event.Logging
+import akka.http.scaladsl.model.HttpMethods._
+import akka.http.scaladsl.model._
+import akka.stream.ActorMaterializer
+import akka.testkit.{ EventFilter, TestProbe }
+import com.typesafe.config.{ Config, ConfigFactory }
+import org.scalatest.{ Matchers, WordSpec }
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class SslConfigWarningsSpec extends WordSpec with Matchers {
+  val testConf: Config = ConfigFactory.parseString("""
+    akka.loglevel = INFO
+    akka.stdout-loglevel = INFO
+    akka.log-dead-letters = OFF
+    akka.http.server.request-timeout = infinite
+
+    akka.ssl-config {
+      loose {
+        disableSNI = true
+        disableHostnameVerification = true
+      }
+    }
+    """)
+
+  "ssl-config loose options should cause warnings to be logged" should {
+
+    "warn if SNI is disabled globally" in {
+      implicit val system = ActorSystem(getClass.getSimpleName, testConf)
+      implicit val materializer = ActorMaterializer()
+
+      val p = TestProbe()
+      system.eventStream.subscribe(p.ref, classOf[Logging.LogEvent])
+
+      EventFilter.warning(start = "Detected that Server Name Indication (SNI) is disabled globally", occurrences = 1) intercept {
+        Http()(system)
+      }
+
+      // the very big warning shall be logged only once per actor system (extension)
+      EventFilter.warning(start = "Detected that Server Name Indication (SNI) is disabled globally", occurrences = 0) intercept {
+        Http()(system)
```

Comment:
Is the warning really logged each time the extension is accessed like this? Otherwise you should probably nest the event filters around one such call.